docs/source/notes/extending.rst

```diff
@@ -233,6 +289,54 @@ See :ref:`grad-check` for more details on finite-difference gradient comparisons
 If your function is used in higher order derivatives (differentiating the backward pass) you
 can use the ``gradgradcheck`` function from the same package to check higher order derivatives.
 
+.. _combining-forward-context:
+
+Combined or separate :meth:`~Function.forward` and ``setup_context``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are two main ways to define :class:`~Function`. Either:
+
+- define a :meth:`~Function.forward` that combines the forward compute logic with ``setup_context``
+- (as of PyTorch 2.0) define a separate :meth:`~Function.forward` and ``setup_context``.
+
+We recommend the second option (separate :meth:`~Function.forward` and ``setup_context``)
+because that is closer to how PyTorch native operations are implemented and it composes
+with :mod:`torch.func` transforms. However, we plan to support both approaches going forward;
+combining :meth:`~Function.forward` with ``setup_context`` leads to more flexibility.
```

Comment:
nit: we aren't super specific about why combining is more flexible here (i.e. allow for saving intermediates?)

Reply:
yeah, saving intermediates without promoting them to be outputs